include/deal.II/dofs/dof_vector.h

```diff
@@ -0,0 +1,243 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2007 - 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__dof_vector_h
+#define dealii__dof_vector_h
+
+#include <deal.II/base/smartpointer.h>
+#include <deal.II/lac/constraint_matrix.h>
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/vector_memory.h>
+
+DEAL_II_NAMESPACE_OPEN
+/**
+ * An object storing a vector describing a finite element function
+ * together with its DoFHandler and the ConstraintMatrix associated
+ * with the handler.
+ *
+ * Data vectors for finite element discretizations do not make sense
+ * without their DoFHandler. Nevertheless, deal.II has assumed for
+ * decades, that there is a single DoFHandler in a program and that
+ * all vectors refer to it. This class encapsulates both, to begin
```

Comment:
The assertion isn't quite true -- several tutorial programs, for example, use multiple DoFHandler objects. In any case, I've found that temporal references (like, "the previous behavior of the library was to...") aren't really useful: users only want to know what the behavior is now. If you want to know what changed, then you look into the changelog; how a class or group of classes behaved is of no interest to those who read a class documentation.